packages/app/src/cli/services/dev/output.test.ts

```diff
@@ -10,6 +16,84 @@ afterEach(() => {
+    }
+
+    test('shows info about tunnel URL and links to Partners Dashboard when app is brand new', async () => {
+      // Given
+      const outputMock = mockAndCaptureOutput()
+      const localApp = await mockApp()
+
+      const remoteApp = testOrganizationApp({newApp: true})
+
+      // When
+      await outputUpdateURLsResult(true, urls, remoteApp, localApp)
+
+      // Then
+      expect(outputMock.output()).toMatch(`For your convenience, we've given your app a default URL:`)
+      expect(outputMock.output()).toMatch(`${urls.applicationUrl}.`)
+      expect(outputMock.output()).toMatch(`You can update your app's URL anytime in the Partners Dashboard [1] But`)
+      expect(outputMock.output()).toMatch(`once your app is live, updating its URL will disrupt user access.`)
+      expect(outputMock.output()).toMatch(
+        `[1] https://partners.shopify.com/${remoteApp.organizationId}/apps/${remoteApp.id}/edit`,
+      )
+    })
+
+    test('shows nothing when urls were updated', async () => {
+      // Given
+      const outputMock = mockAndCaptureOutput()
+      const localApp = await mockApp()
+
+      const remoteApp = testOrganizationApp({newApp: false})
+
+      // When
+      await outputUpdateURLsResult(true, urls, remoteApp, localApp)
+
+      // Then
+      expect(outputMock.output()).toEqual('')
+    })
+
+    test('shows how to update app urls on partners when app is not brand new, urls were not updated and app uses legacy config', async () => {
+      // Given
+      const outputMock = mockAndCaptureOutput()
+      const localApp = await mockApp()
+
+      const remoteApp = testOrganizationApp({newApp: false})
+
+      // When
+      await outputUpdateURLsResult(false, urls, remoteApp, localApp)
+
+      // Then
+      expect(outputMock.output()).toMatch(`To make URL updates manually, you can add the following URLs as redirects`)
+      expect(outputMock.output()).toMatch(`in your Partners Dashboard [1]:`)
+      expect(outputMock.output()).toMatch(`• ${urls.redirectUrlWhitelist[0]}`)
+      expect(outputMock.output()).toMatch(
+        `[1] https://partners.shopify.com/${remoteApp.organizationId}/apps/${remoteApp.id}/edit`,
+      )
+    })
+
+    test('shows how to update app urls with config push when app is not brand new, urls were updated and app uses new config', async () => {
+      // Given
+      const outputMock = mockAndCaptureOutput()
+      const localApp = await mockApp(true)
+
+      const remoteApp = testOrganizationApp({newApp: false})
+
+      // When
+      await outputUpdateURLsResult(false, urls, remoteApp, localApp)
+
+      // Then
+      expect(outputMock.output()).toMatch(`To update URLs manually, add the following URLs to`)
+      expect(outputMock.output()).toMatch(`shopify.app.staging.toml under auth > redirect_urls and run`)
+      expect(outputMock.output()).toMatch(`\`npm run shopify app config push -- --config=shopify.app.staging.toml\``)
```

Comment:
I think here it'd be nice to provide the shorthand, i.e. `npm run shopify app config push -- --config staging`